package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -83,163 +128,201 @@ def get_or_create_modular_pipeline(
-        """Add a child to a modular pipeline.
         Args:
-            modular_pipeline_id: ID of the modular pipeline to add the child to.
-            child: The child to add to the modular pipeline.
-        Example:
-            >>> modular_pipelines = ModularPipelinesRepository()
-            >>> modular_pipeline_child = ModularPipelineChild(
-            ...     id="dataset",
-            ...     type=GraphNodeType.DATA
-            ... )
-            >>> modular_pipelines.add_child("data_science", modular_pipeline_child)
-            >>> data_science_pipeline = modular_pipelines.get_or_create_modular_pipeline(
-            ...     "data_science"
-            ... )
-            >>> assert data_science_pipeline.children == {modular_pipeline_child}
+            modular_pipeline_id (str): The ID of the modular pipeline to add children to.
+            kedro_nodes (List[Node]): The kedro nodes (and it's related datasets/parameters)
+                    to add as children.
         """
         modular_pipeline = self.get_or_create_modular_pipeline(modular_pipeline_id)
-        modular_pipeline.children.add(child)
+        modular_pipeline_inputs_outputs = set(modular_pipeline.inputs) | set(
+            modular_pipeline.outputs
+        )
 
-    def extract_from_node(self, node: GraphNode) -> Optional[str]:
-        """Extract the namespace from a graph node and add it as a modular pipeline node
-        to the modular pipeline repository.
+        for kedro_node in kedro_nodes:
+            # add kedro node as a child to the modular pipeline
+            if kedro_node.namespace == modular_pipeline_id:
+                kedro_node_id = _hash(str(kedro_node))
+                modular_pipeline.children.add(
+                    ModularPipelineChild(id=kedro_node_id, type=GraphNodeType.TASK)
+                )
 
-        Args:
-            node: The GraphNode from which to extract modular pipeline.
-        Returns:
-            ID of the modular pipeline node added to the modular pipeline repository if found.
-        Example:
-            >>> modular_pipelines = ModularPipelinesRepository()
-            >>> model_output_node = GraphNode.create_data_node(
-            ...     "data_science.model_output", layer=None, tags=set(), dataset=None
-            ... )
-            >>> modular_pipelines.extract_from_node(model_output_node)
-            'data_science'
-            >>> assert modular_pipelines.has_modular_pipeline("data_science")
-        """
+                modular_pipeline.tags.update(kedro_node.tags)
 
-        # There is no need to extract modular pipeline from parameters
-        # because all valid modular pipelines are encoded in either a TaskNode or DataNode.
-        if isinstance(node, ParametersNode):
-            return None
+                # add input/output datasets of a kedro node as a child
+                # to the modular pipeline
+                self._add_datasets_as_children(

```

Comment:
I was thinking we could add datasets that have no modular pipeline ID associated with them (and thus belong to the root modular pipeline) in this function instead of in managers.py.